clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -6924,14 +6928,27 @@ void OffloadBundler::ConstructJobMultipleOutputs(
           llvm::sys::path::stem(Input.getFilename()).str() + "-prelink", "o");
     InputFileName = C.addTempFile(C.getArgs().MakeArgString(TmpName));
     LinkArgs.push_back(InputFileName);
-    // Input files consist of fat libraries and the object(s) to be unbundled.
-    for (const auto &I : Inputs)
-      LinkArgs.push_back(I.getFilename());
+    const ToolChain *HTC = C.getSingleOffloadToolChain<Action::OFK_Host>();
+    // Add crt objects
+    LinkArgs.push_back(TCArgs.MakeArgString(HTC->GetFilePath("crt1.o")));
+    LinkArgs.push_back(TCArgs.MakeArgString(HTC->GetFilePath("crti.o")));
     // Add -L<dir> search directories.
     TCArgs.AddAllArgs(LinkArgs, options::OPT_L);
-    for (const auto &A :
-            TCArgs.getAllArgValues(options::OPT_foffload_static_lib_EQ))
-      LinkArgs.push_back(TCArgs.MakeArgString(A));
+
+    // TODO - We can potentially go through the args and add the known linker
+    // pass through args of --whole-archive and --no-whole-archive.  This
+    // would allow for support for usages of -Wl,--whole-archive
+    // -foffload-static-lib=<lib> -Wl,--no-whole-archive
+    // Input files consist of fat libraries and the object(s) to be unbundled.
+    for (const auto &I : Inputs) {
+      if (I.getType() == types::TY_WholeArchive)
+        LinkArgs.push_back("--whole-archive");
+      LinkArgs.push_back(I.getFilename());
+      if (I.getType() == types::TY_WholeArchive)
+        LinkArgs.push_back("--no-whole-archive");
+    }
+    // Add crt objects
+    LinkArgs.push_back(TCArgs.MakeArgString(HTC->GetFilePath("crtn.o")));
```

Comment:
Is there a reason not to do this around line 6934?

Reply:
The ordering here keeps in line with typical link ordering.  crt1/crti at the beginning and crtn at the end.  Just trying to stay consistent here.  I don't know if it really makes a difference in practice, though.